src/packages/cli/src/cli.ts

```diff
@@ -141,6 +148,7 @@ async function startGanache(err: Error) {
       break;
     }
   }
+  versionCheck.init().log();
```

Comment:
This line causes `versionCheck` to kick off some async things in the background; these things could take a while (it currently defaults to 300ms max, but a comment below challenges this default).

What happens if `VersionCheck` is waiting for a response and then Ganache CLI requests that the Node process shuts down? Currently, I believe the Node process will wait until the http2 `request` (and the `session` connection itself, but they are both ended at the same time in the implementation) are both `end`ed.

`VersionCheck` needs a way to be interrupted (something like `versionCheck.destroy()` to end and prevent all network and file system access) _OR_ they whole thing needs to be run in its own orphaned process in the background -- in which case it can take its time. The orphan process is probably a bit too over the top for what we are doing here, so I vote for a syncronous `disconnect` method :-).

Reply:
`getLatestVersion` is not awaited in `init` so any kind of `SIG`, `process.exit`, `ctrl c` that kills the process should treat the request the same as any other unresolved promise and exit.

`conf` does not hold the config file open and uses `atomically` (https://www.npmjs.com/package/atomically) for writing to disk. 


Can you help me to understand the scope. Once a day, within the first `config.TTL` amount of time after start the user or other software using ganache will `SIGINT/QUIT/KILL`, `ctrl c` or `process.exit` and the risk is that the unawaited promise will hold the app open for the remainder of `config.TTL`?

Reply:
> should treat the request the same as any other unresolved promise and exit.

It's the network request that is the problem. We need a longer network timeout (discussed in another comment), but a longer network timeout would keep the process from closing on it's own.

Ganache _does_ (try to ... there is an issues related to this not happening) close cleanly; the process exits on its own because there is no additional work pending in the event loop.

Aside: `process.exit()` should not be used and using it to cleanly exit a process is an anti-pattern (since it means the application didn't cleanup after itself -- so database writes, fs writes, network requests, etc could still be in flight).

Reply:
I believe I understand when this would occur, take a look at the unit test on this line. I believe if you set it and the config `ttl` to `10000` you will see the bug you are describing happen as mocha will be held open until the TTL expires even though the tests have completed running.
https://github.com/trufflesuite/ganache/blob/68314da400ad8a87774b7c4356802023e0c0877e/src/packages/version-check/tests/index.test.ts#L1164

Even if the `ttl` of VersionCheck was low the test would still hold open because it runs the server with the response delay. But I'm able to setup an env just breaking the test server on into a separate process, setting the `ttl` high and pointing the test at the server.

I agree a `destroy` the closes the `req` and `session/client` is the way. 

This has a unit test but was not hit against a high `ttl` https://github.com/trufflesuite/ganache/blob/68314da400ad8a87774b7c4356802023e0c0877e/src/packages/version-check/tests/index.test.ts#L1273

Reply:
:+1: `destroy` function added that closes the request/session. 
https://github.com/trufflesuite/ganache/pull/3285/commits/234c81689c529bb19278f57a25bfaab25f2a8701

Reply:
As part of cli's shutdown process versionCheck's `destroy` should be called now.

Reply:
@davidmurdoch Any thoughts on how I'm doing it [here](https://github.com/trufflesuite/ganache/pull/3285/files#diff-d5cc68b0e026fee131b90a36b9f5bb25b719c702a557cb904ecde02dda59f78cR79)?

I wasn't wild about keeping the VC around when it can be garbage collected after it runs.